### PR TITLE
Revert #3107 + Bump to k8s v1.18.0 and minikube v1.19.2

### LINKED ISSUE #3122: kubernetes 1.18, kubectl apply --validate=yes can't install CRD
**What happened?**
kubectl apply --validate=yes can't install CRD from github URL

I try to run following command
```
kubectl --namespace=prometheus apply --validate=true -f https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagers.yaml
```
and got exit code 1
and following error
```
The CustomResourceDefinition "alertmanagers.monitoring.coreos.com" is invalid:
* spec.validation.openAPIV3Schema.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
* spec.validation.openAPIV3Schema.properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
```

**Did you expect to see something different?**
yes, on kubectl 1.17.x and kubernetes 1.17, I didn't get any errors

**How to reproduce it (as minimally and precisely as possible)**:
- install k8s 1.18 from zero
- try install latest CRD's with validation

**Environment**
clear installation 
minikube 1.9.1 + kubernetes 1.18

* Kubernetes version information:

    `kubectl version`
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:50:46Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:

    minikube

### PR DESCRIPTION
**Pull request by simonpasquier**
Revert #3107 which prevented the creation of the Prometheus Operator's CRDs on Kubernetes >= v1.18.0. As a consequence, this PR bumps the end-to-end tests to use Kubernetes v1.18.0.

Closes #3122

### CODE REVIEW

File: scripts/create-minikube.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@ set -u
 # print each command before executing it
 set -x
 
-export MINIKUBE_VERSION=v1.2.0
-export KUBERNETES_VERSION=v1.14.5
+export MINIKUBE_VERSION=v1.9.2
+export KUBERNETES_VERSION=v1.18.0
```
Comment (lilic) [inline scripts/create-minikube.sh:12]: There was 1.18.1 is that already available to be used?

File: scripts/create-minikube.sh [hunk 2]
```diff
@@ -34,20 +34,34 @@ minikube config set WantUpdateNotification false
 minikube config set WantReportErrorPrompt false
 minikube config set WantNoneDriverWarning false
 minikube config set vm-driver none
+minikube config view
+
+# Setup Docker according to https://kubernetes.io/docs/setup/production-environment/container-runtimes/#docker
```
Comment (lilic) [inline scripts/create-minikube.sh:39]: Why do we need to setup docker? Do we use it for building images?

File: scripts/create-minikube.sh (hunk 1 above)
Comment (simonpasquier) [reply scripts/create-minikube.sh]: updated

File: scripts/create-minikube.sh (hunk 2 above)
Comment (simonpasquier) [reply scripts/create-minikube.sh:39]: Minikube deploys the kubernetes components using Docker (`vm-driver=none`).

**Pull request merged**